packages/ifpack2/src/Ifpack2_Details_Fic_def.hpp

```diff
@@ -0,0 +1,287 @@
+  computeTime = localPrec->getComputeTime();
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+bool Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+isComputed() const
+{
+  return computedFlag;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+Teuchos::RCP<const Tpetra::RowMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> >
+Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getMatrix() const
+{
+  return mat;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+int Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumInitialize() const
+{
+  return nInit;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+int Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumCompute() const
+{
+  return nComputed;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+int Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getNumApply() const
+{
+  return nApply;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+double Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getInitializeTime() const 
+{
+  return initTime;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+double Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getComputeTime() const
+{
+  return computeTime;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+double Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+getApplyTime() const
+{
+  return applyTime;
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+void Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+checkLocalIC()
+{
+  localPrec->checkIC();
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+void Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+printStatus()
+{
+  std::cout << "FAST-ILU Preconditioner" << std::endl;
+  std::cout << "IC - symmetric" << std::endl;
+  std::cout << "n = " << mat->getNodeNumRows() << std::endl;
+  std::cout << "nnz = " << mat->getNodeNumEntries() << std::endl;
+  std::cout << "sweeps = " << localPrec->getNFact() << std::endl;
+  std::cout << "t init = " << initTime << "s" << std::endl; 
+  std::cout << "t comp = " << computeTime << "s" << std::endl; 
+  std::cout << "t appl = " << applyTime << "s" << std::endl; 
+  localPrec->checkIC();
+}
+
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+void Fic<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+setMatrix(const Teuchos::RCP<const TRowMatrix>& A)
+{
+  //bmk note: this modeled after RILUK::setMatrix
+  if(mat.get() != A.get())
+  {
+    mat = A;
+    //note: initFlag not cleared because initialize() doesn't need to be called again if A's structure didn't change
```

Comment:
In order to know whether A's structure didn't change, you would need to check that `A.getGraph().getRawPtr() == mat.getGraph().getRawPtr()`.  Otherwise, the structure must have changed, since `A` and `mat` are totally different objects.